internal/yamlalias/wrap_test.go

```diff
@@ -0,0 +1,149 @@
+package yamlalias
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func TestWrap(t *testing.T) {
+	type Simple struct {
+		Foo string `yaml:"foo" yaml-aliases:"bar,baz"`
+	}
+
+	type ManyTypes struct {
+		Str   string            `yaml-aliases:"string"`
+		Int   int               `yaml-aliases:"Int"`
+		Slice []string          `yaml-aliases:"list"`
+		Map   map[string]string `yaml-aliases:"dict"`
+	}
+
+	type AliasToPtr struct {
+		Foo *string `yaml:"foo" yaml-aliases:"bar"`
+	}
+
+	tests := []struct {
+		msg  string
+		data string
+		want interface{}
+	}{
+		{
+			msg:  "No aliases",
+			data: `{"foo": "foo", "bar": "bar"}`,
+			want: &struct {
+				Foo string
+				Bar string
+			}{"foo", "bar"},
+		},
+		{
+			msg:  "set field by normal name",
+			data: `{"foo": "foo"}`,
+			want: &Simple{"foo"},
+		},
+		{
+			msg:  "set field by first alias",
+			data: `{"bar": "bar"}`,
+			want: &Simple{"bar"},
+		},
+		{
+			msg:  "set field by second alias",
+			data: `{"baz": "baz"}`,
+			want: &Simple{"baz"},
+		},
+		{
+			msg:  "set field with multiple aliases, last one wins",
+			data: `{"bar": "bar", "foo": "foo", "baz": "baz"}`,
+			want: &Simple{"baz"},
+		},
+		{
+			msg:  "pointer field that is not set",
+			data: `{}`,
+			want: &AliasToPtr{},
+		},
+		{
+			msg:  "pointer field set directly",
+			data: `{"foo": "foo"}`,
+			want: &AliasToPtr{stringPtr("foo")},
+		},
+		{
+			msg:  "pointer field that is set by alias",
+			data: `{"bar": "bar"}`,
+			want: &AliasToPtr{stringPtr("bar")},
+		},
```

Comment:
What happens if the pointer field is already pointing to another value?
(This is probably its own test outside this table.)

```
var dest string
out := AliasToPtr{Foo: &dest}
yamlalias.Unmarshal([]byte(`{"bar": "bar"}`), &out)
```

Reply:
Added a new test `TestWrapPointerField`, `dest` is updated through the alias as well.